### PR TITLE
[Merged by Bors] - Add smesherID to logs around ATX building and for supervised service

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

To easily distinguish logs for different smeshers in the ATX building process.
## Description

Note: the smesherID in the supervised service logs is not necessary in production (there is always only 1) but it is helpful in debugging tests (as there might be many).

## Test Plan

n/a

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: activation/activation.go [hunk 1]
```diff
@@ -405,7 +406,11 @@ func (b *Builder) BuildNIPostChallenge(ctx context.Context, nodeID types.NodeID)
 	case err != nil:
 		return nil, fmt.Errorf("get nipost challenge: %w", err)
 	case challenge.PublishEpoch < current:
-		// challenge is stale
+		logger.Info(
+			"existing NiPOST challenge is stale, resetting state",
+			zap.Stringer("current_epoch", current),
```
Comment (dshulyak) [inline activation/activation.go]: i think it is better to use zap.Uint32 . it is cheaper to encode and you can also do filtering (>, <) in loki

**Review by dshulyak — APPROVED** [review]

**Review by fasmat — APPROVED** [review]

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**